src/internal/connector/exchange/exchange_data_collection.go

```diff
@@ -257,6 +257,14 @@ func (col *Collection) streamItems(ctx context.Context) {
 					break
 				}
 
+				// If the data is no longer available just return here and chalk it up
+				// as a success. There's no reason to retry and no way we can backup up
+				// enough information to restore the item anyway.
+				if e := graph.IsErrDeletedInFlight(err); e != nil {
+					atomic.AddInt64(&success, 1)
```

Comment:
Let's add a log here, so that we can easily identify it during supportability concerns.  Info level should be fine.